Criterion/Report.hs

```diff
@@ -185,58 +170,6 @@ formatReport reports templateName = do
         fmap TextBlock (f (T.unpack fp))
     includeNode _ n = return n
 
-    -- Merge Report with it's analysis and outliers
-    merge :: ToJSON a => a -> Value -> Value
-    merge x y = case toJSON x of
-        Object x' -> case y of
-            Object y' -> Object (x' <> y')
-            _         -> y
-        _         -> y
-
-    inner :: Report -> IO Value
-    inner r@Report {..} = do
-      reportName' <- sanitizeJSString $ T.pack reportName
-      return $ merge reportAnalysis $ merge reportOutliers $ object
-        [ "name"                  .= reportName'
-        , "json"                  .= TLE.decodeUtf8 (encode r)
-        , "number"                .= reportNumber
-        , "iters"                 .= vector "x" iters
-        , "times"                 .= vector "x" times
-        , "cycles"                .= vector "x" cycles
-        , "kdetimes"              .= vector "x" kdeValues
-        , "kdepdf"                .= vector "x" kdePDF
-        , "kde"                   .= vector2 "time" "pdf" kdeValues kdePDF
-        , "anMeanConfidenceLevel" .= anMeanConfidenceLevel
-        , "anMeanLowerBound"      .= anMeanLowerBound
-        , "anMeanUpperBound"      .= anMeanUpperBound
-        , "anStdDevLowerBound"    .= anStdDevLowerBound
-        , "anStdDevUpperBound"    .= anStdDevUpperBound
-        ]
-      where
-        [KDE{..}]          = reportKDEs
-        SampleAnalysis{..} = reportAnalysis
-
-        iters  = measure measIters reportMeasured
-        times  = measure measTime reportMeasured
-        cycles = measure measCycles reportMeasured
-        anMeanConfidenceLevel
-               = confidenceLevel $ confIntCL $ estError anMean
-        (anMeanLowerBound, anMeanUpperBound)
-               = confidenceInterval anMean
-        (anStdDevLowerBound, anStdDevUpperBound)
-               = confidenceInterval anStdDev
-
-        sanitizeJSString :: T.Text -> IO T.Text
```

Comment:
The removal of `sanitizeJSString` makes me worried that we're accidentally reverting the fix for #224. How does this branch render the program in https://github.com/bos/criterion/issues/224#issue-611029139?

Reply:
We're not injecting any JavaScript any more. Instead a `data-report-json` attribute is used in an HTML tag which is then read and parsed with `JSON.parse`. This is precisely to avoid sanitizing and escaping javascript manually which is really tricky to get right.